Source/Core/Core/FifoPlayer/FifoRecorder.cpp

```diff
@@ -224,3 +226,117 @@ FifoRecorder& FifoRecorder::GetInstance()
+  ProcessVertexComponent(ARRAY_POSITION, vtx_desc.low.Position, offset, vertex_size, num_vertices,
+                         vertex_data);
+  offset += pos_size;
+
+  const u32 norm_size =
+      VertexLoader_Normal::GetSize(vtx_desc.low.Normal, vtx_attr.g0.NormalFormat,
+                                   vtx_attr.g0.NormalElements, vtx_attr.g0.NormalIndex3);
+  ProcessVertexComponent(ARRAY_NORMAL, vtx_desc.low.Position, offset, vertex_size, num_vertices,
+                         vertex_data);
+  offset += norm_size;
+
+  for (u32 i = 0; i < vtx_desc.low.Color.Size(); i++)
+  {
+    const u32 color_size =
+        VertexLoader_Color::GetSize(vtx_desc.low.Color[i], vtx_attr.GetColorFormat(i));
+    ProcessVertexComponent(ARRAY_COLOR0 + i, vtx_desc.low.Position, offset, vertex_size,
+                           num_vertices, vertex_data);
+    offset += color_size;
+  }
+  for (u32 i = 0; i < vtx_desc.high.TexCoord.Size(); i++)
+  {
+    const u32 tc_size = VertexLoader_TextCoord::GetSize(
+        vtx_desc.high.TexCoord[i], vtx_attr.GetTexFormat(i), vtx_attr.GetTexElements(i));
+    ProcessVertexComponent(ARRAY_TEXCOORD0 + i, vtx_desc.low.Position, offset, vertex_size,
+                           num_vertices, vertex_data);
+    offset += tc_size;
+  }
+
+  ASSERT(offset == vertex_size);
+}
+
+// If a component is indexed, the array it indexes into for data must be saved.
+void FifoRecorder::FifoRecordAnalyzer::ProcessVertexComponent(int array_index,
+                                                              VertexComponentFormat array_type,
+                                                              u32 component_offset, u32 vertex_size,
+                                                              u16 num_vertices,
+                                                              const u8* vertex_data)
+{
+  // Skip if not indexed array
+  if (!IsIndexed(array_type))
+    return;
+
+  u16 max_index = 0;
+
+  // Determine min and max indices
+  if (array_type == VertexComponentFormat::Index8)
+  {
+    for (u16 vertex_num = 0; vertex_num < num_vertices; vertex_num++)
+    {
+      const u8 index = vertex_data[component_offset];
+      vertex_data += vertex_size;
+
+      // 0xff skips the vertex
+      if (index != 0xff)
+      {
+        if (index > max_index)
+          max_index = index;
+      }
+    }
+  }
+  else
+  {
+    for (u16 vertex_num = 0; vertex_num < num_vertices; vertex_num++)
+    {
+      const u16 index = Common::swap16(&vertex_data[component_offset]);
+      vertex_data += vertex_size;
+
+      // 0xffff skips the vertex
+      if (index != 0xffff)
+      {
+        if (index > max_index)
+          max_index = index;
+      }
+    }
+  }
+
+  const u32 array_start = m_cpmem.array_bases[array_index];
+  const u32 array_size = m_cpmem.array_strides[array_index] * (max_index + 1);
+
+  FifoRecorder::GetInstance().UseMemory(array_start, array_size, MemoryUpdate::VERTEX_STREAM);
```

Comment:
Just thinking out aloud here.

This is going to result in creating individual memory update for every single triangle strip or other primitive group out of the same vertex buffer. 

It might be smarter to simply cache the min_index and max_index from function and make the memory as used the next time CPmem is updated to point at new vertex buffers.